src/scene_battle_rpg2k3.cpp

```diff
@@ -2083,6 +2089,65 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 		FaceTarget(*actor, *action->GetTarget());
 	}
 
+	if (action->GetCBAMovement() != lcf::rpg::BattlerAnimationItemSkill::Movement_none) {
+		SetBattleActionState(BattleActionState_MoveForward);
+	} else {
+		SetBattleActionState(BattleActionState_StartAlgo2);
+	}
+	return BattleActionReturn::eContinue;
+}
+
+Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleActionMoveForward(Game_BattleAlgorithm::AlgorithmBase* action) {
+	auto* source = action->GetSource();
+
+	if (source->GetType() == Game_Battler::Type_Ally) {
+		auto* target = action->GetTarget();
+
+		if (cba_move_frame == 0) {
+			auto* actor = static_cast<Game_Actor*>(source);
+			auto* sprite = actor->GetActorBattleSprite();
+			if (sprite) {
+				sprite->SetAnimationState(Sprite_Actor::AnimationState_WalkingLeft);
```

Comment:
Does this work correctly if the target is to the right of the attacker? (Back attack etc..)

Reply:
This one works because the sprites are mirrored on back attack. For setting the `cba_end_pos` variable there is even a check if the sprite is mirrored.